From f22de27e548f743c7741ab5df0328f82a7768362 Mon Sep 17 00:00:00 2001
From: Haoran Wang <b50027@freescale.com>
Date: Mon, 27 Jul 2015 08:42:51 +0800
Subject: [PATCH] MA-6859 bcmdhd: Use core sdio_reset_comm

sdio_reset_comm() called to reset/re-enumerate
SDIO interface.  If this is not called then SDIO
device will have "disappeared" and come up
as "not found".

Use the sdio_reset_comm in core driver will fix
the unavailable update firmware in ENABLE_INSMOD_NO_FW_LOAD
mode.

Signed-off-by: Haoran Wang <b50027@freescale.com>
---
 drivers/net/wireless/bcmdhd/Makefile       | 2 +-
 drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c | 8 +-------
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index 88ab641..a49b5b9 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -11,7 +11,7 @@ DHDCFLAGS += -Wall -Wstrict-prototypes -Dlinux -DLINUX -DBCMDRIVER            \
 	-DEMBEDDED_PLATFORM -DPNO_SUPPORT                                     \
 	-DDHD_DONOT_FORWARD_BCMEVENT_AS_NETWORK_PKT                           \
 	-DCUSTOMER_HW2 -DGET_CUSTOM_MAC_ENABLE	\
-	-DCONFIG_DTS
+	-DCONFIG_DTS -DENABLE_INSMOD_NO_FW_LOAD
 
 #################
 # Common feature
diff --git a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c
index 22cef80..181cf47 100644
--- a/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c
+++ b/drivers/net/wireless/bcmdhd/bcmsdh_sdmmc.c
@@ -59,7 +59,7 @@ static void IRQHandler(struct sdio_func *func);
 static void IRQHandlerF2(struct sdio_func *func);
 #endif /* !defined(OOB_INTR_ONLY) */
 static int sdioh_sdmmc_get_cisaddr(sdioh_info_t *sd, uint32 regaddr);
-static int sdio_reset_comm(struct mmc_card *card);
+extern int sdio_reset_comm(struct mmc_card *card);
 
 #define DEFAULT_SDIO_F2_BLKSIZE		512
 #ifndef CUSTOM_SDIO_F2_BLKSIZE
@@ -92,12 +92,6 @@ DHD_PM_RESUME_WAIT_INIT(sdioh_request_buffer_wait);
 
 int sdioh_sdmmc_card_regread(sdioh_info_t *sd, int func, uint32 regaddr, int regsize, uint32 *data);
 
-/* workaround to simply return 0 as suggested by Broadcom */
-static int sdio_reset_comm(struct mmc_card *card)
-{
-	return 0;
-}
-
 static int
 sdioh_sdmmc_card_enablefuncs(sdioh_info_t *sd)
 {
-- 
1.8.0

